libsolidity/parsing/Parser.cpp

```diff
@@ -1244,7 +1244,8 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 	{
 		expectToken(Token::New);
 		ASTPointer<TypeName> typeName(parseTypeName(false));
-		nodeFactory.setEndPositionFromNode(typeName);
+		if (typeName)
+			nodeFactory.setEndPositionFromNode(typeName);
```

Comment:
`else nodeFactory.markEndPosition()` ?

Reply:
it marks some end position automatically

Reply:
ah sorry, I'm wrong there